src/robusta/core/discovery/discovery.py

```diff
@@ -54,9 +61,49 @@ class Config:
         arbitrary_types_allowed = True
 
 
+DISCOVERY_PIPE = "/tmp/discovery_pipe"
+SHUTDOWN_SIGNAL = "shutdown"
+STACKDUMP_SIGNAL = "stackdump"
+
+
 class Discovery:
     executor = ProcessPoolExecutor(max_workers=1)  # always 1 discovery process
 
+    @staticmethod
+    def init_discovery_error_pipes():
+        try:
+            os.mkfifo(DISCOVERY_PIPE)
+        except FileExistsError:
+            pass
+        except Exception:
+            logging.error(
+                "Failed to create discovery pipes",
+                exc_info=True,
+            )
+
+    @staticmethod
+    def send_signal_to_pipe(signal):
+        try:
+            with open(DISCOVERY_PIPE, mode="w") as pipe:
+                logging.info("Sending signal to discovery thread")
+                pipe.write(signal)
```

Comment:
also check that this cant be blocking (waiting for the other side of the pipe to be open)